packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -300,11 +311,15 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
     return this.context.getState();
   }
   /** Dispose of the current CallWithChatAdapter. */
-  public dispose(): void {
-    this.chatAdapter.offStateChange(this.onChatStateChange);
+  public async dispose(): Promise<void> {
+    this.chatAdapterPromise.then((adapter) => {
```

Comment:
these are async functions, use the async await syntax and not the older promise syntax, i.e.
```ts
public async dispose(): Promise<void> {
    const chatAdapter = await this.chatAdapterPromise;
    chatAdapter.offStateChange(this.onChatStateChange);
    chatAdapter.dispose();

    this.callAdapter.offStateChange(this.onCallStateChange);
    this.callAdapter.dispose();
  }
```

Do the same with all of these functions